lib/puppet/functions/azure_key_vault/lookup.rb

```diff
@@ -7,18 +9,22 @@
       vault_name => String,
       vault_api_version => String,
       Optional[metadata_api_version] => String,
+      Optional[onprem_agent_api_version] => String,
       confine_to_keys => Array[String],
       Optional[key_replacement_token] => String,
-      Optional[service_principal_credentials] => String
+      Optional[service_principal_credentials] => String,
+      Optional[return_sensitive_type] => Boolean
```

Comment:
This module is "safe by default".  What that means is this function was designed and tested to always return a type of "Sensitive" to the calling code in order to prevent accidental leakage of secrets in logs.  Curious as to why you need to allow a string to be returned.

Reply:
yep. But in our case we need that option 'return_sensitive_type' so we can disable 'safe' :) we need strings.
but as default that value is false

we need to be able to do this:
```
/hiera.yaml
- name: 'Azure Key Vault Secrets'
    lookup_key: azure_key_vault::lookup
    options:
      vault_name: production-vault
      vault_api_version: '7.3'
      onprem_agent_api_version: '2021-02-01'
      return_sensitive_type: false
      key_replacement_token: '-'
      confine_to_keys:
        - '^prod.*'

/data/nodes/perconadb57.percona.zzz.yaml
root: "%{lookup('percona-root-password')}"
```

Reply:
we're using a lof of old puppet modules that does not work fine with sensitive type :( so we can't override their values via hiera yaml files

Reply:
we try to solve this :) https://github.com/TraGicCode/tragiccode-azure_key_vault/issues/105

Reply:
Hey @marcelmitto ,

My recommendation would to be to follow the roles and profiles pattern similiar to this:





Resources: